stix_shifter/src/json_to_stix/json_to_stix_translator.py

```diff
@@ -138,20 +140,35 @@ def transform(self, obj):
             'objects': {}
         }
 
+        if self.hash_options:
+            hash_types, log_source_id_map, generic_hash_name, hash_type_values = qradar_utils.parse_hash_options(self.hash_options, obj)
         # create normal type objects
         for ds_key in obj:
             if ds_key not in ds_map:
                 logging.debug('{} is not found in map, skipping'.format(ds_key))
                 continue
+
+            generic_hash_key = ''
+            # Check and handle file hash of unknown type
+            if self.hash_options:
+                try:
+                    generic_hash_key = qradar_utils.lookup_hash_with_logsource_id(
```

Comment:
the callback idea is that `def transform(self, obj):` would become `def transform(self, obj, callback):`
and then
```
generic_hash_key = callback(...)
```